indico/modules/events/ical.py

```diff
@@ -95,38 +101,46 @@ def generate_event_component(event, user=None):
     return component
 
 
-def event_to_ical(event, user=None, detailed=False):
+def event_to_ical(event, user=None, scope=None):
     """Serialize an event into an ical.
 
     :param event: The event to serialize
     :param user: The user who needs to be able to access the events
-    :param detailed: If True, iCal will include the event's contributions
+    :param scope: If True, iCal will include a detailed timetable up to this event scope
     """
-    return events_to_ical([event], user, detailed)
+    return events_to_ical([event], user, scope)
 
 
-def events_to_ical(events, user=None, detailed=False):
+def events_to_ical(events, user=None, scope=None):
     """Serialize multiple events into an ical.
 
     :param events: A list of events to serialize
     :param user: The user who needs to be able to access the events
-    :param detailed: If True, iCal will include the event's contributions
+    :param scope: If True, iCal will include a detailed timetable up to this event scope
     """
     calendar = icalendar.Calendar()
     calendar.add('version', '2.0')
     calendar.add('prodid', '-//CERN//INDICO//EN')
 
     for event in events:
-        if not detailed:
+        if not scope or not CalendarScope.get('scope'):
```

Comment:
Not sure what the second check is meant for (i guess the quotes weren't intended), but I don't think it's necessary - we always expect none or an enum member.

Maybe the whole nested if/else construct here could be unnested to something like this?

```python
if scope == CalendarScope.X:
    ....
elif scope == CalendarScope.Y:
    ....
else:
    ...
```

The duplicated code would be minimal (just the loop over components).

Reply:
Yes, this was stupid, the reason was that shortly before pushing I was using strings as scope, then I decided to update it to an Enum.